pkgs/applications/virtualization/libnvidia-container/default.nix

```diff
@@ -7,6 +7,7 @@
 , libseccomp
 , rpcsvc-proto
 , libtirpc
+, nvidia_x11
```

Comment:
I'm not sure this is the right way to go. For all other CUDA related packages we try hard to not directly link against linuxPackages.nvidia_x11. The issue here is that if you run non-NixOS Linux your kernel version will most likely not match whatever is currently linuxPackages.
Also for NixOS this solution is not ideal as it breaks using different channels for different packages.

I'm sorry that I don't have an alternative solution, but I just wanted to point out that the `/run/` solution is there for good reason.

Reply:
Thank you for raising your concerns! I was quite sure that my solution was not the ideal one :). 

We could add a comment to the source code, manifesting your concerns?

For myself, I am happy to have a running solution for this problem and think others might be too maybe

Reply:
I'm assuming you are trying to fix this on NixOS. Can't we have the nvidia-docker NixOS module create that `/run/nvidia-docker/bin` symlink you mention to be missing?

Reply:
Hm. So this seems to be done by `virtualisation.docker.enableNvidia`.
Are you using that setting?

Reply:
yap. that option is on. 
I think I had mentioned previously in another issue (https://github.com/NixOS/nixpkgs/issues/131608) that I would very much welcome if somebody managed to generate the missing /run/nvidia-docker/bin symlinks. I myself did not find how to do it in the code..

Reply:
Hm. That's somewhat confusing. This option should generate these directories.
https://github.com/NixOS/nixpkgs/blob/ed4771d6e23001d55bf6b422699ad798dcd61d77/nixos/modules/hardware/video/nvidia.nix#L336
Maybe you have conflicting definitions in your configuration.nix?
Could you maybe share a minimal example to reproduce the issue?
As far as I understand this should already work and makes this workaround less attractive.

Reply:
I'm somewhat busy these weeks and will try to get back to you.

Reply:
I found the issue in the file you referenced (nvidia.nix). I leave a comment in the general discussion of this PR.

Reply:
Well, maybe this is not the right way to go but that's something what works. Nvidia support for docker is just broken for years, maybe it would be a good idea to accept the solution which is not perfect but works?..